File: OOP/Lab5/C#/BaseClass.cs
using System;

namespace Lab5_OOP
{
    internal class bString
    {
        protected string[] data { get; set; }

        internal bString(params string[] input)
        {
            data = input;
        }

        public int getsLength(int index)
        {
            if ((index < 0) || (index > data.Length))
            {
                Console.WriteLine("Index out of range.");
                return -1;
            }

            return data[index].Length;
        }

        public int getLength() => data.Length;
    }
}

File: OOP/Lab5/C++/BaseClass.cpp
#include "BaseClass.h"
#include<iostream>

using namespace std;

bString::bString(string input[]){
    data = input;
}

int bString::getLength(int index){
    if (index >= data->length() ){
        cout << "Your index is out of range.";
        return -1;
    }
    return data[index].length();
}